rsconnect/actions.py

```diff
@@ -1144,25 +1148,25 @@ def _gather_basic_deployment_info_for_framework(
     if new and app_id:
         raise api.RSConnectException("Specify either a new deploy or an app ID but not both.")
 
-    if app_id is not None:
-        # Don't read app metadata if app-id is specified. Instead, we need
-        # to get this from Connect.
-        app = api.get_app_info(connect_server, app_id)
-        app_mode = AppModes.get_by_ordinal(app.get("app_mode", 0), True)
-
-        logger.debug("Using app mode from app %s: %s" % (app_id, app_mode))
-
-    if not new and app_id is None:
+    app_mode = AppModes.STATIC
```

Comment:
The `app_mode` parameter should contain the correct value for the framework... I'm not sure I understand why this is set to `STATIC`.